lib/db-utils.js

```diff
@@ -3,18 +3,33 @@ const sqlite3 = require('sqlite3')
 
 const p = (obj, fn) => promisify(obj[fn].bind(obj))
 
-const mapKeys = (obj, fn) => Object.keys(obj).reduce((copy, key) => {
-  copy[fn(key, obj[key])] = obj[key]
-  return copy
-}, {})
+const noop = p => p
+
+const mapObject = (obj, keyMapFn, valMapFn = noop) =>
+  Object.keys(obj).reduce((copy, key) => {
+    copy[keyMapFn(key, obj[key])] = valMapFn(obj[key])
+    return copy
+  }, {})
+
+const isNotValue = (val) => {
+  const [key] = Object.keys(val)
+  return key === '$not'
```

Comment:
This could be achieved using `Object.prototype.hasOwnProperty` instead of having to get all the keys

Reply:
perfect!